packages/atomic/src/components/search/atomic-result-html/atomic-result-html.spec.ts

```diff
@@ -0,0 +1,213 @@
+import type {Result} from '@coveo/headless';
+import {html} from 'lit';
+import {ifDefined} from 'lit/directives/if-defined.js';
+import {beforeEach, describe, expect, it, vi} from 'vitest';
+import {renderInAtomicResult} from '@/vitest-utils/testing-helpers/fixtures/atomic/search/atomic-result-fixture';
+import {renderInAtomicSearchInterface} from '@/vitest-utils/testing-helpers/fixtures/atomic/search/atomic-search-interface-fixture';
+import {buildFakeSearchEngine} from '@/vitest-utils/testing-helpers/fixtures/headless/search/engine';
+import {buildFakeResult} from '@/vitest-utils/testing-helpers/fixtures/headless/search/result';
+import type {AtomicResultHtml} from './atomic-result-html';
+import './atomic-result-html';
+
+vi.mock('@coveo/headless', {spy: true});
+
+describe('atomic-result-html', () => {
+  let mockResult: Result;
+  const mockedEngine = buildFakeSearchEngine();
+
+  const locators = {
+    getHtml: (element: AtomicResultHtml) =>
+      element?.querySelector('atomic-html'),
+  };
+
+  beforeEach(async () => {
+    mockResult = buildFakeResult({
+      title: 'Test <b>HTML</b> Title',
+      raw: {
+        custom_html_field: '<p>Custom <em>HTML</em> content</p>',
+        plain_field: 'Plain text content',
+        empty_field: '',
+        null_field: null,
+      },
+    });
+  });
+
+  const renderResultHtml = async ({
+    props = {},
+    result = mockResult,
+  }: {
+    props?: Partial<{field: string; sanitize: boolean}>;
+    result?: Result | null;
+  } = {}) => {
+    const {element} = await renderInAtomicResult<AtomicResultHtml>({
+      template: html`<atomic-result-html
+        field=${ifDefined(props.field)}
+        ?sanitize=${props.sanitize ?? true}
+      ></atomic-result-html>`,
+      selector: 'atomic-result-html',
+      result: result === null ? undefined : result,
+      bindings: (bindings) => {
+        bindings.engine = mockedEngine;
+        return bindings;
+      },
+    });
+
+    return {
+      element,
+    };
+  };
+
+  describe('when added to the DOM', () => {
+    it('should not set the error when the field prop is a non-empty string', async () => {
+      const {element} = await renderResultHtml({props: {field: 'title'}});
+
+      expect(element.error).toBeUndefined();
+    });
+
+    it('should set the error when the field prop is an empty string', async () => {
+      const {element} = await renderResultHtml({props: {field: ''}});
+
+      expect(element.error).toBeInstanceOf(Error);
+      expect(element.error.message).toMatch(/field: value is an empty string/i);
+    });
+
+    it('should set the error when the field prop is undefined', async () => {
+      const {element} = await renderResultHtml();
+
+      expect(element.error).toBeInstanceOf(Error);
+      expect(element.error.message).toMatch(/field: value is required/i);
+    });
+  });
+
+  describe('when the field property changes', () => {
```

Comment:
in my opinion, test line from 82-112 is not necessary
it's not the input like UI input that submit to database, so basically you set/change in the code is the same